src/App.tsx

```diff
@@ -1,26 +1,498 @@
 /* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  addTodos,
+  getTodos,
+  deleteTodo,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<ErrorType>(ErrorType.DEFAULT_VALUE);
+  const [filter, setFilter] = useState<FilterType>(FilterType.ALL);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
```

Comment:
```suggestion
  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
```
it can be one state.